src/Analyser/ResultCache/ResultCacheManager.php

```diff
@@ -317,6 +319,11 @@ private function isMetaDifferent(array $cachedMeta, array $currentMeta): bool
 			$currentMeta['projectConfig'] = Neon::encode($currentMeta['projectConfig']);
 		}
 
+		// Do not invalidate result cache generated before introducing support for result cache meta extensions
```

Comment:
This is not necessary. Any new PHPStan version invalidates the entire result cache.